### PR TITLE
379, 380 Updates to sponsor data

### LINKED ISSUE #379: Mandatory sponsor data
make it mandatory for sponsor to add company link, web description and web logo when applying for sponsorship.

### PR DESCRIPTION
**Pull request by dpoirier**
* Make required:
  * web logo
  * web description
  * external URL

* Move web logo and web description to be fields on Sponsor instead of
  separate SponsorBenefit records

* Remove the now-obsolete web logo and web description benefit types,
  and a couple others that aren't used anymore.

* Remove:
  * print description

* Add optional fields:
  * Booth number
  * Job fair table number
  * Promo codes

* Add the new optional fields to those that can be interpolated when
  sending bulk email to sponsors.

Fixes #379, #380

### CODE REVIEW

File: pycon/sponsorship/models.py [hunk 1]
```diff
@@ -71,11 +62,11 @@ def sponsors(self):
 
 class Sponsor(models.Model):
 
-    applicant = models.ForeignKey(User, related_name="sponsorships", verbose_name=_("applicant"), null=True)
+    applicant = models.ForeignKey(User, related_name="sponsorships", verbose_name=_("applicant"), null=True, on_delete=SET_NULL)
 
     name = models.CharField(_("Sponsor Name"), max_length=100)
-    display_url = models.URLField(_("display URL"), blank=True)
-    external_url = models.URLField(_("external URL"))
+    display_url = models.URLField(_("display URL - text to display on link to sponsor page, if different from the actual link"), blank=True)
```
Comment (rebecca-caktus) [inline pycon/sponsorship/models.py]: Would this be better named `display_name` or somesuch?

Comment (dpoirier) [inline pycon/sponsorship/models.py]: Yes, it would. I guess I could try renaming this existing field.

File: pycon/sponsorship/models.py [hunk 2]
```diff
@@ -172,13 +178,7 @@ def joblisting_text(self):
...
     @property
     def website_logo(self):
-        if self.sponsor_logo is None:
-            benefits = self.sponsor_benefits.filter(benefit__type="weblogo", upload__isnull=False)[:1]
-            if benefits.count():
-                if benefits[0].upload:
-                    self.sponsor_logo = benefits[0]
-                    self.save()
-        return self.sponsor_logo.upload
+        return self.weblogo
```
Comment (rebecca-caktus) [inline pycon/sponsorship/models.py]: typo?

File: pycon/sponsorship/tests/test_forms.py [hunk 3]
```diff
@@ -6,6 +9,10 @@
 
 from .factories import SponsorLevelFactory
 
+# Tiny image file for testing
+TEST_IMAGE_FILENAME = os.path.join(os.path.dirname(__file__), 'colormap.gif')
```
Comment (rebecca-caktus) [inline pycon/sponsorship/tests/test_forms.py]: I don't know that it matters, but you could generate this sort of file in a utility function. I personally prefer that to creating a particular file.

File: pycon/sponsorship/forms.py [hunk 4]
```diff
@@ -46,8 +53,13 @@ class Meta:
         fields = [
             "name",
             "external_url",
+            "display_url",
+            "web_description",
+            "web_logo",
             "contact_name",
             "contact_emails",
+            "contact_phone",
+            "contact_address",
```
Comment (rebecca-caktus) [inline pycon/sponsorship/forms.py]: I just noticed that the field order differs between the application form and the details/edit form. Would it be possible to combine these forms, or at least inherit one from the other?

File: pycon/sponsorship/models.py (hunk 2 above)
Comment (dpoirier) [inline pycon/sponsorship/models.py]: On Thu, Jul 23, 2015 at 11:46 AM, Rebecca <notifications@github.com> wrote:

> > -        return self.sponsor_logo.upload
> > +        return self.weblogo
>
> typo?
>

​Thanks, you have sharp eyes.​

File: pycon/sponsorship/forms.py (hunk 4 above)
Comment (dpoirier) [inline pycon/sponsorship/forms.py]: On Thu, Jul 23, 2015 at 12:09 PM, Rebecca <notifications@github.com> wrote:

> Would it be possible to combine these forms, or at least inherit one from
> the other?


​Yes!  Good idea.​



*Dan Poirier*Developer

dpoirier@caktusgroup.com
www.caktusgroup.com

**Comment by dpoirier** [comment]
@rebecca-caktus updated

**Comment by rebecca-caktus** [comment]
:moneybag:

**Pull request merged**